packages/cactus-plugin-ledger-connector-besu/src/test/typescript/benchmark/run-plugin-ledger-connector-besu-benchmark.ts

```diff
@@ -0,0 +1,239 @@
+    rpcApiHttpHost,
+    rpcApiWsHost,
+    pluginRegistry,
+    logLevel,
+  };
+  const connector = new PluginLedgerConnectorBesu(options);
+  pluginRegistry.add(connector);
+
+  const expressApp = express();
+  expressApp.use(bodyParser.json({ limit: "250mb" }));
+  const server = http.createServer(expressApp);
+
+  const wsApi = new SocketIoServer(server, {
+    path: Constants.SocketIoConnectionPathV1,
+  });
+
+  const listenOptions: IListenOptions = {
+    hostname: "127.0.0.1",
+    port: 0,
+    server,
+  };
+  const addressInfo = (await Servers.listen(listenOptions)) as AddressInfo;
+  test.onFinish(async () => await Servers.shutdown(server));
+  const { address, port } = addressInfo;
+  const apiHost = `http://${address}:${port}`;
+  // t.comment(
+  //   `Metrics URL: ${apiHost}/api/v1/plugins/@hyperledger/cactus-plugin-ledger-connector-besu/get-prometheus-exporter-metrics`,
+  // );
+
+  // const wsBasePath = apiHost + Constants.SocketIoConnectionPathV1;
+  // t.comment("WS base path: " + wsBasePath);
+  const besuApiClientOptions = new BesuApiClientOptions({ basePath: apiHost });
+  const apiClient = new BesuApiClient(besuApiClientOptions);
+
+  await installOpenapiValidationMiddleware({
+    logLevel,
+    app: expressApp,
+    apiSpec: OAS,
+  });
+
+  await connector.getOrCreateWebServices();
+  await connector.registerWebServices(expressApp, wsApi);
+
+  return {
+    httpApi: apiClient,
+    apiServer: connector,
+  };
+};
+
+const main = async (opts: { readonly argv: Readonly<Array<string>> }) => {
+  const logLevel: LogLevelDesc = "WARN";
+
+  const gitRootPath = path.join(
+    __dirname,
+    "../../../../../../", // walk back up to the project root
+  );
+
+  console.log("%s gitRootPath=%s", LOG_TAG, gitRootPath);
+
+  const DEFAULT_OUTPUT_FILE_RELATIVE_PATH =
+    ".tmp/benchmark-results/plugin-ledger-connector-besu/run-plugin-ledger-connector-besu-benchmark.ts.log";
+
+  const relativeOutputFilePath =
+    opts.argv[2] === undefined
+      ? DEFAULT_OUTPUT_FILE_RELATIVE_PATH
+      : opts.argv[2];
+
+  console.log(
+    "%s DEFAULT_OUTPUT_FILE_RELATIVE_PATH=%s",
+    LOG_TAG,
+    DEFAULT_OUTPUT_FILE_RELATIVE_PATH,
+  );
+
+  console.log("%s opts.argv[2]=%s", LOG_TAG, opts.argv[2]);
+
+  console.log("%s relativeOutputFilePath=%s", LOG_TAG, relativeOutputFilePath);
+
+  const absoluteOutputFilePath = path.join(gitRootPath, relativeOutputFilePath);
+
+  console.log("%s absoluteOutputFilePath=%s", LOG_TAG, absoluteOutputFilePath);
+
+  const absoluteOutputDirPath = path.dirname(absoluteOutputFilePath);
+  console.log("%s absoluteOutputDirPath=%s", LOG_TAG, absoluteOutputDirPath);
+
+  await fse.mkdirp(absoluteOutputDirPath);
+  console.log("%s mkdir -p OK: %s", LOG_TAG, absoluteOutputDirPath);
+
+  const { apiServer, httpApi } = await createTestInfrastructure({ logLevel });
+
+  const minSamples = 100;
+  const suite = new Benchmark.Suite({});
+
+  const cycles: string[] = [];
+
+  await new Promise((resolve, reject) => {
```

Comment:
Setup of output directories and running the benchmark logic could be moved to separate functions